### DISCUSSION TITLE
What happened when I select Economical option in creating Knowledge?

### DISCUSSION THREAD

**Discussion by huy302** [langgenius/dify#6978] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

The only hint I got is "Use offline vector engines, keyword indexes, etc. to reduce accuracy without spending tokens". It does not mention what vector db or embedding algorithm is being used? Anyone can shed some light on this?

Thanks

**Comment by huy302** [comment]
any help here? judging from the CPU processing time after uploading I guess it's building a local vector db, but not sure what kind, thanks

**Reply by crazywoola** [reply]
https://github.com/langgenius/dify/blob/28d4e5b04503fd8c351b754ef58e0fc65d09cfe9/api/core/rag/retrieval/dataset_retrieval.py#L535

See this method.
